### PR TITLE
Use V5 database to get list of images for migration

### PR DESCRIPTION
**Pull request by arnaud-morvan**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -42,20 +42,13 @@ Building and running with Docker
 Launch images migration from V5 to V6
 -------------------------------------
 
-Migration takes source images from v5 S3 read-only bucket, which could be
-defined by environment variables, example:
+Migration get list of images from v5 database which could be defined by
```
Comment (tsauerwein) [inline README.md]: "The migration retrieves a list of images from the v5 database. The connection to the database can be defined as environment variable, for example:"

**Comment by arnaud-morvan** [comment]
@asaunier @tsauerwein : This is needed to get images working correctly on demov6 with S3 backend, if no one has comments, I will merge it today.

File: README.md [hunk 2]
```diff
@@ -26,6 +26,64 @@ At that time, a request is sent to image backend to move original and resized
...
 image is associated with an authenticated user.
 
+Configuration
+-------------
+
+Configuration should be set by environment variables:
+
+``STORAGE_BACKEND``: (required) ``s3`` or ``local``
+
+* ``s3``: require ``INCOMING_FOLDER`` and ``ACTIVE_FOLDER``, should be used in
```
Comment (tsauerwein) [inline README.md]: "requires"

File: README.md [hunk 3]
```diff
@@ -26,6 +26,64 @@ At that time, a request is sent to image backend to move original and resized
...
+  production.
+* ``local``: require ``INCOMING_BUCKET`` and ``ACTIVE_BUCKET``, should be used
```
Comment (tsauerwein) [inline README.md]: "requires"

File: c2corg_images/scripts/migrate.py [hunk 4]
```diff
@@ -1,45 +1,89 @@
...
+        batch_size = 500
+        offset = 0
+        connection = self.v5_engine.connect()
+        sql = """
+SELECT count(*) AS count
+FROM app_images_archives;
+"""
+        count_result = connection.execute(sql)
+        for row in count_result:
+            total = row['count']
```
Comment (tsauerwein) [inline c2corg_images/scripts/migrate.py]: You could write this as:

```
total = connection.execute(sql).fetchone()[0]
```

File: c2corg_images/scripts/migrate.py [hunk 5]
```diff
@@ -1,45 +1,89 @@
...
+        while offset < total:
+            sql = """
+SELECT filename
+FROM app_images_archives
+LIMIT {} OFFSET {};
+""".format(batch_size, offset)
+
+            result = connection.execute(sql)
+            for row in result:
+                yield row['filename']
```
Comment (tsauerwein) [inline c2corg_images/scripts/migrate.py]: :+1: for using generators!

File: c2corg_images/scripts/migrate.py [hunk 6]
```diff
@@ -1,45 +1,89 @@
...
+
+            offset += batch_size
+        connection.close()
```
Comment (tsauerwein) [inline c2corg_images/scripts/migrate.py]: You could use the `with` statement to implicitly close the connection.

File: c2corg_images/scripts/migrate.py [hunk 7]
```diff
@@ -1,45 +1,89 @@
...
-            with open(temp_storage.object_path(key), 'wb') as fd:
-                for chunk in r.iter_content(None):
-                    fd.write(chunk)
-        else:
-            v5_storage.copy(key, temp_storage)
+        tries = 3
+        success = False
+        while tries > 0 and not success:
+            try:
+                r = requests.get('http://s.camptocamp.org/uploads/images/{}'.
```
Comment (tsauerwein) [inline c2corg_images/scripts/migrate.py]: It probably makes sense to set a connection timeout: http://docs.python-requests.org/en/master/user/quickstart/#timeouts

File: c2corg_images/scripts/migrate.py (hunk 4 above)
Comment (arnaud-morvan) [inline c2corg_images/scripts/migrate.py]: >       total = connection.execute(sql).fetchone()[0]
E       AttributeError: 'list' object has no attribute 'fetchone'

So I will use : 
```
total = connection.execute(sql)[0]['count']
```

**Pull request merged**